sdks/python/apache_beam/runners/worker/bundle_processor.py

```diff
@@ -199,26 +199,19 @@ def finish(self):
 
 
 class _StateBackedIterable(object):
-  def __init__(self, state_handler, state_key, coder_or_impl):
+  def __init__(self, state_handler, state_key, coder_or_impl,
+               is_cached=False):
     self._state_handler = state_handler
     self._state_key = state_key
     if isinstance(coder_or_impl, coders.Coder):
       self._coder_impl = coder_or_impl.get_impl()
     else:
       self._coder_impl = coder_or_impl
+    self._is_cached = is_cached
 
   def __iter__(self):
-    # This is the continuation token this might be useful
-    data, continuation_token = self._state_handler.blocking_get(self._state_key)
-    while True:
-      input_stream = coder_impl.create_InputStream(data)
-      while input_stream.size() > 0:
-        yield self._coder_impl.decode_from_stream(input_stream, True)
-      if not continuation_token:
-        break
-      else:
-        data, continuation_token = self._state_handler.blocking_get(
-            self._state_key, continuation_token)
+    return self._state_handler.blocking_get(
```

Comment:
Is there a JIRA for the continuation token that can be referenced here?

Reply:
Could you elaborate? The continuation token is respected as before.

Reply:
For the continuation token logic that was removed, what is the replacement?

Reply:
There was no continuation token logic removed. Robert raised the point that the state is materialized no matter how many continuation tokens are available (only if Caching is enabled). This can result in too large state being loaded. However, neither is the continuation token used by the Runner currently, nor is this likely to happen for user state, rather for side inputs which are currently not cached.

Reply:
The code that is removed keeps fetching the state until there is no token. My question was what the replacement for this is? If you are saying that there was no need for it in first place because there is no support on the runner side, then that's OK. It might be appropriate to capture this and the question raised by Robert in a comment and/or link to the appropriate JIRA.

Reply:
>The code that is removed keeps fetching the state until there is no token.

This code has _not_ been removed. It has simply been refactored to support fetching all state at once instead of just one element at a time. The logic regarding the continuation token is unchanged. Please look inside the `materialize_iter` method.

Of course I will address all remaining comments before merging the PR, as soon as I get a chance.

Reply:
Added a comments to further clarify.